it/utils_for_tests.go

```diff
@@ -142,22 +148,14 @@ func IsBundleResourceCond(t *testing.T, namespace, name string, resource smith_v
 	}
 }
 
-func IsBundleNewerCond(namespace, name string, resourceVersions ...string) watch.ConditionFunc {
+func IsBundleObservedGenerationCond(namespace, name string) watch.ConditionFunc {
 	return func(event watch.Event) (bool, error) {
 		metaObj := event.Object.(meta_v1.Object)
 		if metaObj.GetNamespace() != namespace || metaObj.GetName() != name {
 			return false, nil
 		}
 		b := event.Object.(*smith_v1.Bundle)
-		for _, rv := range resourceVersions {
-			if b.ResourceVersion == rv {
-				// TODO Should be using Generation here once it is available
-				// https://github.com/kubernetes/kubernetes/issues/7328
-				// https://github.com/kubernetes/features/issues/95
-				return false, nil
-			}
-		}
-		return true, nil
+		return b.Status.ObservedGeneration >= b.Generation, nil
```

Comment:
Just FYI: CRDs don't get `Generation` incremented when annotations/labels updated, see https://github.com/kubernetes/kubernetes/issues/67428
I haven't checked if `ResourceVersion` get incremented in that case, but just something to keep in mind.

Reply:
> Just FYI: CRDs don't get Generation incremented when annotations/labels updated, see kubernetes/kubernetes#67428

Yes, and that is what I would expect. See https://github.com/atlassian/smith/issues/325 and the corresponding PR. Just found these docs https://github.com/kubernetes/community/blob/master/contributors/design-proposals/api-machinery/customresources-subresources.md#status-behavior and https://github.com/kubernetes/kubernetes/issues/45539

> I haven't checked if ResourceVersion get incremented in that case, but just something to keep in mind.

Yes, it does. It changes when a mutating request touches `/status` or `/` (normal resource path). I tested =)